### PR TITLE
Add documentation for custom evaluators

### PR DESCRIPTION
**Pull request by rock3r**
This PR adds the documentation for the custom evaluators feature (see #45).

### CODE REVIEW

File: docs/incubating/custom-evaluator.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Table of contents
+ * [The `DefaultViolationsEvaluator`](#the-defaultviolationsevaluator)
+ * [Creating a custom violations evaluator](#creating-a-custom-violations-evaluator)
+
+---
+
+## The `DefaultViolationsEvaluator`
+The plugin has a default mechanism to decide whether to consider a build as passing or failed. The mechanism is manifesting itself
+as the `penalties` closure:
```
Comment (mr-archano) [inline docs/incubating/custom-evaluator.md]: it's actually `penalty`

File: docs/incubating/custom-evaluator.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+staticAnalysis {
+    penalties {
+        maxErrors = 0
+        maxWarnings = 10
+    }
+    //...
+}
+```
+
+This closure instructs the plugin to use a [`DefaultViolationsEvaluator`][defaultviolationsevaluatorcode] that will count the number of
```
Comment (mr-archano) [inline docs/incubating/custom-evaluator.md]: It's worth noting that the default evaluator ships with a default configuration, that is you don;t need to specify to define `penalty{}` and you will roll with `DefaultViolationsEvaluator` using the default thresholds (see [`PenaltyExtension.groovy#L5-L6`](https://github.com/novoda/gradle-static-analysis-plugin/blob/master/plugin/src/main/groovy/com/novoda/staticanalysis/PenaltyExtension.groovy#L5-L6))

File: docs/incubating/custom-evaluator.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+[Configurable failure thresholds](../advanced-usage.md#configurable-failure-thresholds) documentation.
+
+## Creating a custom violations evaluator
+In order to provide a custom evaluator, you can implement the [`ViolationsEvaluator`][violationsevaluatorcode] interface and provide
+that implementation to the `evaluator` property of the `staticAnalysis` closure. The [`ViolationsEvaluator`][violationsevaluatorcode]
+can be provided as a closure as well:
+
+```gradle
+staticAnalysis {
+	evaluator {	Set<Violations> allViolations ->
```
Comment (tasomaniac) [inline docs/incubating/custom-evaluator.md]: This piece of code uses tabs 😄 Also before `Set` we have tabs, maybe we should have a single space there.

File: docs/incubating/custom-evaluator.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+The evaluation logic can be any arbitrary function that respects this contract:
+ * The evaluator receives a set containing all the [`Violations`][violationscode] that have been collected by the tools (one per tool)
+ * If the build is to be considered successful, then the evaluator will run to completion without throwing exceptions
+ * If the build is to be considered failed, then the evaluator will throw a `GradleException`
+
+Anything that respect such contract is valid. For example, a custom evaluator might:
+ * Collect all the report files and upload them somewhere, or send them to Slack or an email address
+ * Use the GitHub API to report the issues on the PR that the build is running on, à la [GNAG](https://github.com/btkelly/gnag)
+ * Only break the build if there's errors or warnings in one specific report
```
Comment (tasomaniac) [inline docs/incubating/custom-evaluator.md]: `there are`?

File: docs/incubating/custom-evaluator.md (hunk 1 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: I _knew_ I'd misspell it LOL

File: docs/incubating/custom-evaluator.md (hunk 3 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: I copy-pasted it from Antonio's PR! I swear :D But nice catch, will fix

Comment (tasomaniac) [reply docs/incubating/custom-evaluator.md]: @mr-archano prefers tabs over spaces. 

~[](https://media.giphy.com/media/4ORDH8n5cdPA4/giphy.gif)

File: docs/incubating/custom-evaluator.md (hunk 1 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: cf03f2b9e4c71d7df49f7a07111715bb0ee96957

File: docs/incubating/custom-evaluator.md (hunk 3 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: 17d17d2d010d77b9215fb4345dc52f7dfc6079c8

File: docs/incubating/custom-evaluator.md (hunk 4 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: adbcdc21179b35d402a9707265ed00aa337a144a

**Comment by rock3r** [comment]
@tasomaniac I thought about providing a sample, but couldn't come up with anything meaningful that could fit in a couple of lines. In the meantime, giving a couple of ideas in writing seems a good starting point to me, but feel free to suggest something. I just want to avoid a lengthy example as this is documentation and not a sample project.

**Comment by tasomaniac** [comment]
@rock3r I think the ideas section is really helpful. I agree that the sample would not contain much meaningful code. It would just showcase the available options. 

Maybe something like this? 

```groovy
staticAnalysis {
    evaluator { Set<Violations> allViolations ->
        allViolations.each {
            println "Name of the tool: $it.name Errors: $it.errors Warnings: $it.warnings"
        }
    }
    //...
}
```

File: docs/incubating/custom-evaluator.md (hunk 2 above)
Comment (rock3r) [reply docs/incubating/custom-evaluator.md]: f41c03f7144286431a651708a3f949390e4304ab

**Comment by rock3r** [comment]
@tasomaniac yeah but it doesn't seem particularly meaningful to me :) showing the closure gets 90% of the information across anyway

**Pull request merged**